app/controllers/homes_controller.rb

```diff
@@ -1,4 +1,7 @@
 class HomesController < ApplicationController
+  before_action :authenticate_user!
+
   def index
+    @posts = Post.where(status: 1)
```

Comment:
With the definition in post.rb `enum status: [:awaiting_approval, :approval, :disapproval]`
then you must use these enum, e.g: `@posts = Post.approval`, or @post.update_attributes(status: :awaiting_approval), or `@post.awaiting_approval!` to update an enum column

Please find and read document carefully when want to apply new things
https://api.rubyonrails.org/v5.2.3/classes/ActiveRecord/Enum.html